internal/network/manifests_generator.go

```diff
@@ -509,3 +511,71 @@ func NewConfig() (*Config, error) {
 	}
 	return &networkCfg, nil
 }
+
+const nodeIpHint = `
+apiVersion: machineconfiguration.openshift.io/v1
+kind: MachineConfig
+metadata:
+  labels:
+    machineconfiguration.openshift.io/role: {{.ROLE}}
+  name: 10-{{.ROLE}}s-node-ip-hint
+spec:
+  config:
+    ignition:
+      config: {}
+      security:
+        tls: {}
+      timeouts: {}
+      version: 2.2.0
+    networkd: {}
+    passwd: {}
+    storage:
+      files:
+      - contents:
+          source: data:text/plain;charset=utf-8;base64,{{.NODE_IP_CONTENT}}
+          verification: {}
+        filesystem: root
+        mode: 420
+        path: /etc/default/nodeip-configuration
+`
+
+// Add node ip hint is supported from 4.11 and will allow us to tell to node-ip script which ip kubelet should run with
+// https://github.com/openshift/machine-config-operator/commit/a0c9a3caa54018eb89eb5bdd6ec1b8fbf97f6fb7
+func (m *ManifestsGenerator) AddNodeIpHint(ctx context.Context, log logrus.FieldLogger, cluster *common.Cluster) error {
+	if hintSupported, err := common.VersionGreaterOrEqual(cluster.OpenshiftVersion, "4.11.0-0.alpha"); err != nil {
```

Comment:
I'm tempted to say that we should backport this PR to 2.5 [when the 4.10 backport lands](https://github.com/openshift/machine-config-operator/pull/3058).

What do you think about making this check `>4.10` ? or just rem,moving it entirely? 

Creating the `/etc/default/nodeip-configuration` file should be harmless for older versions.

Reply:
i have no issue with changing it to 4.10 when we will have it in 4.10

Reply:
but if there is nothing that can break in 4.10 , we can do it too

Reply:
changed